sklearn/ensemble/bagging.py

```diff
@@ -256,6 +255,35 @@ def fit(self, X, y, sample_weight=None):
         self : object
             Returns self.
         """
+        return self._fit(X, y, self.max_samples, sample_weight)
+
+    def _fit(self, X, y, max_samples, sample_weight=None):
```

Comment:
Is there reasons not to do this in IsolationForest only?

It looks bad from api view.